### PR TITLE
stuff

### PR DESCRIPTION
**Pull request by franz**
* fixes a bunch of milestone 1.0 bugs
* makes SPIR handling more consistent (by not pretending to accept SPIR when cl_khr_spir is not in extension list)
* update documentation (i suspect this will need more work)

### CODE REVIEW

File: doc/sphinx/source/features.rst [hunk 1]
```diff
@@ -18,8 +18,7 @@ Frontend/Clang
   * pipes (WIP)
   * device-side enqueue
 
-* cl_khr_f16: half precision float literals
-
+* cl_khr_f16: half precision support (with the exception of  vload_half / vstore_half)
```
Comment (pjaaskel) [inline doc/sphinx/source/features.rst]: What was the problem with vloads?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1199,7 +1210,7 @@ if (OCS_AVAILABLE)
   add_subdirectory("examples")
 endif()
 
-add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND} ${COMMAND_USES_TERMINAL})
+add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND} -L "'internal|amdsdk_30|piglit|PyOpenCL|conformance_suite_micro'" -j ${CORECOUNT}  ${COMMAND_USES_TERMINAL})
```
[the remark this answers is not in the document]
Comment (franz) [reply CMakeLists.txt]: OK, i'll make it two separate targets

File: doc/sphinx/source/features.rst (hunk 1 above)
Comment (franz) [reply doc/sphinx/source/features.rst]: The problem is not with vload halfs, the problem is with everything else (this is the list of unsupported features). Basically halfs on CPU are completely unsupported with the exception of vload_half and vstore_half.

Comment (pjaaskel) [reply doc/sphinx/source/features.rst]: Ah, right.

**Pull request merged**